RAGchain/preprocess/text_splitter/token_splitter.py

```diff
@@ -0,0 +1,103 @@
+    You can specify a window_size and overlap_size to split the document into overlapping passages.
+    """
+    def __init__(self, tokenizer_name: str = 'tiktoken', chunk_size: int = 0, chunk_overlap: int = 0, **kwargs):
+        """
+        :param tokenizer_name: A tokenizer_name name. You can choose tokenizer_name.
+                        (tiktoken, spaCy, SentenceTransformers, NLTK, huggingFace)
+        :param chunk_size: Maximum size of chunks to return. Default is 0.
+        :param chunk_overlap: Overlap in characters between chunks. Default is 0.
+        :param kwargs: Additional arguments.
+        All splitters were inherited TextSplitter class in langchain text_splitter.py.
+        """
+        self.chosen_tokenizer = tokenizer_name
+
+        # tiktoken
+        self.tiktoken_splitter = TokenTextSplitter.from_tiktoken_encoder(chunk_size = chunk_size, chunk_overlap =  chunk_overlap)
+
+        # spaCy
+        self.spaCy_splitter = SpacyTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # SentenceTransformers (Default: chunk_overlap=0)
+        self.SentenceTransformers_splitter = SentenceTransformersTokenTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # NLTK
+        self.NLTK_splitter = NLTKTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # Hugging Face
+        tokenizer = GPT2TokenizerFast.from_pretrained("gpt2")
+        self.huggingFace_splitter = CharacterTextSplitter.from_huggingface_tokenizer(
+            tokenizer, chunk_size= chunk_size, chunk_overlap= chunk_overlap
+        )
+
+
+
+    def split_document(self, document: Document) -> List[Passage]:
+        """
+        Split a document.
+        """
+        global split_texts, split_documents
+
+        doc_copy = copy.deepcopy(document)
+
+        if self.chosen_tokenizer == 'tiktoken':
+            split_texts = self.tiktoken_splitter.split_text(document.page_content)
+            split_documents = self.tiktoken_splitter.create_documents(split_texts)
+
+        elif self.chosen_tokenizer == 'spaCy':
+            split_texts = self.spaCy_splitter.split_text(document.page_content)
+            split_documents = self.spaCy_splitter.create_documents(split_texts)
+
+        elif self.chosen_tokenizer == 'SentenceTransformers':
+            split_texts = self.SentenceTransformers_splitter.split_text(document.page_content)
+            split_documents = self.SentenceTransformers_splitter.create_documents(split_texts)
+
+        elif self.chosen_tokenizer == 'NLTK':
+            split_texts = self.NLTK_splitter.split_text(document.page_content)
+            split_documents = self.NLTK_splitter.create_documents(split_texts)
+
+        elif self.chosen_tokenizer == 'huggingFace':
+            split_texts = self.huggingFace_splitter.split_text(document.page_content)
+            split_documents = self.huggingFace_splitter.create_documents(split_texts)
+
```

Comment:
Plus, this code is quite repetitive. 
Why don't you make function for this? 
The function's input will be chosen splitter, and return split_documents.